claimManagement/src/main/java/org/openimis/imisclaims/network/request/PostNewClaimRequest.java

```diff
@@ -0,0 +1,209 @@
+package org.openimis.imisclaims.network.request;
+
+import androidx.annotation.NonNull;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openimis.imisclaims.domain.entity.PendingClaim;
+import org.openimis.imisclaims.network.dto.IdentifierDto;
+import org.openimis.imisclaims.network.exception.InsertClaimException;
+import org.openimis.imisclaims.util.DateUtils;
+
+import java.util.Objects;
+
+public class PostNewClaimRequest extends BaseFHIRPostRequest<PendingClaim, String> {
```

Comment:
My main concern with its implementation is that it takes of a lot of concerns as it needs to know the internals of many domain objects. As you did with your DTOs, I'd extract all the conversion in dedicated interface to isolate their procressing knowledge.

Reply:
There is already a lot of logic in the DTO regarding how to transform it from the original JSON.
It seemed to me that putting this logic here made the most sense as I didn't forsee reusing any of this logic anywhere else.

Reply:
we discussed that live on Friday, it's fine by me, especially that we don't want to invest that much on the present code base.